internal/server/Server.go

```diff
@@ -152,3 +152,28 @@ func (s *Server) UpdateGameStates() {
 		agent.UpdateGameState(gs)
 	}
 }
+
+// had to override to address the fact that agents only have access to the game dump
+// version of agents, so if the recipients are set to be those it will panic as they
+// can't call the handler functions
+func (s *Server) RunMessagingSession() {
```

Comment:
Yes well spoted. That does make things tricky but this looks like it should work